### PR TITLE
clang-tidy: run through performance checks

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: hardware/Limitless.cpp [hunk 1]
```diff
@@ -167,10 +167,6 @@ CLimitLess::CLimitLess(const int ID, const int LedType, const int BridgeType, co
 	Init();
 }
 
-CLimitLess::~CLimitLess(void)
-{
```
Comment (gizmocuz) [inline hardware/Limitless.cpp:171]: I added this back, each class should have a deconstructor in .cpp file as it's easier for possible additions
I have no problem with a empty one

File: main/SQLHelper.cpp [hunk 2]
```diff
@@ -3332,7 +3332,7 @@ void CSQLHelper::ManageExecuteScriptTimeout(int pid, int timeout, bool *stillRun
 }
 #endif
 
-void CSQLHelper::PerformThreadedAction(const _tTaskItem tItem)
+void CSQLHelper::PerformThreadedAction(const _tTaskItem &tItem)
```
Comment (gizmocuz) [inline main/SQLHelper.cpp]: This is where it crashed last time, please do not undo my changes
Also we do not need to modify the taskItem after the function returns

File: hardware/Limitless.cpp (hunk 1 above)
Comment (neheb) [reply hardware/Limitless.cpp:171]: The goal is size reduction.

File: main/SQLHelper.cpp (hunk 2 above)
Comment (neheb) [reply main/SQLHelper.cpp]: This was automated. I'll look into it.

Comment (neheb) [reply main/SQLHelper.cpp]: Removed. Easier than figuring out why clang-tidy thinks this should be a reference.

**Comment by gizmocuz** [comment]
Thank you!!

**Pull request merged**